sadx-dc-lighting/lantern.cpp

```diff
@@ -15,6 +15,8 @@
 #include "datapointers.h"
 #include "lantern.h"
 
+DataArray(ColorPair[256], LSPAL, 0x3B12210, 10);
```

Comment:
I notice there's a discrepancy here between the number of palette indices in this array compared to `LanternInstance::palette_index_count`. Will this become a problem? Or if those last two indices are used, do we know what for?

Reply:
The 9th one is used for temporary palette backups when the game blends palettes (e.g. the lights turning off and back on in Gamma's briefing). The 10th one is unused, at least in the code we have access to. Since those extra pairs aren't stored in the PL file, `palette_index_count` 8 is correct.

Reply:
I've looked at what the DC version does with these arrays. On DC it backs up and restores both diffuse 0/specular 0 and diffuse 1/specular 1, while on PC it only copies and restores diffuse 0/specular 0. So that 10th index is also used for palette backups on DC. However, the backups are never cleared on DC. Also, on DC these two extra palette pairs are not in the same place, while on PC they're at the bottom of the regular palette list.

I changed the array size to 8 to imitate the original behavior so these backup palettes are no longer cleared when new PL files are loaded.

Reply:
I'm OK with leaving the array size at 10 and just clearing the first 8, if you wanna go that route. In either case, a code comment describing this behavior would be great.

Reply:
Okay. I changed the array length back to 10 and added a comment with an explanation. The extra palettes still aren't cleared like on DC, which I think is fine considering it's the game's original behavior.